File: hotalg/wish/L206.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None or head.next == None:
            return head
        else:
            a = head
            b = head.next
            a.next = None

            while b != None:
                c = b.next
                b.next = a
                a = b
                b = c
            return a

File: algorithm/src/main/java/linshen/link/L206.java
package linshen.link;

import sulqn.ListNode;

public class L206 {
    public ListNode reverseList(ListNode head) {
        if(head == null)return null;
        ListNode tail = head;
        ListNode cur = head.next;
        tail.next = null;
        while (cur != null) {
            ListNode x = cur.next;
            cur.next = tail;
            tail = cur;
            cur = x;
        }
        return tail;
    }
}
